File: Domain/Entities/Course.cs
namespace Domain.Entities
{
    public class Course : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Introduction { get; set; } = string.Empty;
        public string ImageCourse {  get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public double Price { get; set; }
        public int TotalCourseDuration { get; set; }
        public int NumberOfStudent {  get; set; }
        public int NumberOfPurchases { get; set; }
        public DateTime UpdateTime {  get; set; }
        public string Requirements { get; set; }

        public Guid CreatorId { get; set; }
        public User? Creator { get; set; }
        public virtual ICollection<Subject>? Subjects { get; set; }
        public virtual ICollection<RegisterStudy>? RegisterStudies { get; set; }
        public virtual ICollection<Bill>? Bills { get; set; }
    }
}

File: client/src/app/model/course.ts
import { User } from "./user";

export interface Course {
  id: string;
  name: string;
  introduction: string;
  imageCourse: string;
  code: string;
  price: number;
  totalCourseDuration: number;
  numberOfStudent: number;
  numberOfPurchases: number;
  creator: User;
  updateTime: Date;
  requirement: string;
  subjects: any;

  // creator: User;
  // public virtual ICollection<Subject>? Subjects { get; set; }
  // public virtual ICollection<RegisterStudy>? RegisterStudies { get; set; }
  // public virtual ICollection<Bill>? Bills { get; set; }
}
